src/app/countdown/directives/countdown.directive.ts

```diff
@@ -0,0 +1,44 @@
+import { Directive, EventEmitter, Input, Output } from '@angular/core';
+import * as moment from 'moment';
+import { Subject } from 'rxjs';
+import { switchMap, take, takeUntil, tap } from 'rxjs/operators';
+
+import { Destroyable } from '../../util/destroyable';
+import { SecondService } from '../services/second.service';
+
+@Directive({
+  selector: '[expDateToSeconds]'
+})
+export class CountdownDirective extends Destroyable {
+
+  private dateFeed$: Subject<string> = new Subject();
+
+  @Input() set expDateToSeconds(value: string) {
+    this.dateFeed$.next(value);
```

Comment:
storing this input into a subject lets us switchMap cancel the previous subscription. It's cleaner imo than managing the subscriptions ourselves.